src/vm_memory_monitor.erl

```diff
@@ -225,7 +239,8 @@ start_link(MemFraction, AlarmSet, AlarmClear) ->
                           [MemFraction, {AlarmSet, AlarmClear}], []).
 
 init([MemFraction, AlarmFuns]) ->
-    TRef = start_timer(?DEFAULT_MEMORY_CHECK_INTERVAL),
+    ets:new(?MODULE, [named_table, public]),
```

Comment:
Would it be preferable to use the process dictionary instead of a public ets table?

Reply:
There can (and usually will) be multiple processes that call the function.

Reply:
Since we're using 1 second cache expiration time it makes sense to keep the value in the process state. The process is updating its state every second anyway. The question is do we want it to be 1 second or we need a better resolution?

Reply:
@michaelklishin that makes sense. I suppose we don't want to serialize operations via `call` within this `gen_server`. I brought it up because I can only find 2 other instances of public `ets` tables in the `stable` source code.

Reply:
@hairyhum 1 second sounds perfectly fine. We can use process dictionary here and maybe that would avoid a contention point with a large number of active queues. @hairyhum @dumbbell @dcorbacho WDYT?

Reply:
@michaelklishin process dictionary would be fine in this case, it's only a tiny amount of info.

Reply:
@hairyhum any objections to refactoring this to use process dictionaries?

Reply:
Why do we need process dictionary, but not the gen_server state? All usages of this function call the server to get memory limit (`get_memory_limit()` function) anyway.

Reply:
> Why do we need process dictionary, but not the gen_server state?

Using the state would work well too. I didn't think of that.